### PR TITLE
Feature/usercourses

### PR DESCRIPTION
**Pull request by dav1dchang**
# What's here (& why)

- (include link(s) to applicable Trello card(s))

# Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete / work-in-progress, PR is for discussion or feedback
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes 
- [x] There are no merge conflicts

### CODE REVIEW

**Review by chazkiker2 — CHANGES_REQUESTED** [review]
These changes look solid so far. 

## Changes to make: 

- Delete the following files: `TeacherRepository`, `StudentRepository`, `TeacherModelAssembler`, `StudentModelAssembler`

## Couple of things to consider. 

We can hop in discord to figure these out together, but wanted to write them down before I forgot about them

- We may want to keep `StudentController` and `TeacherController`, respectively as to abstract all student-specific vs teacher-specific logic out of our `UserController` in general.
- We may want to add in more role-based checks throughout these changes. But that would require getting our `UserRole` up and running which I'm thinking about tackling next.

## TLDR

Either way, these considerations would apply to future PRs... I actually think this PR should be good to go after we delete those files

**Comment by chazkiker2** [comment]
@dav1dchang, since you're AFK currently and the changes I requested are tiny, I committed the changes so we can get this great PR into main.

**Review by chazkiker2 — APPROVED** [review]
Deleted `StudentRepository` and `TeacherRepository` so we can use your awesome changes throughout the app.

**Comment by BUGBOUNTYchrisg8691** [comment]
Does this need to be reviewed and merged?

**Comment by chazkiker2** [comment]
Yessir @BUGBOUNTYchrisg8691

**Review by BUGBOUNTYchrisg8691 — APPROVED** [review]
looks good
